### PR TITLE
feat: MLIBZ-2430 Item isn't deleted from the backend after update delete operations with store type.sync

### PR DESCRIPTION
**Pull request by yuliya-guseva**
#### Description
Item isn't deleted from the backend after update delete operations with store type.sync.

Precondition: 
Create dataStore with StoreType.SYNC

Steps:
1. Create item (state1) and push it to the server 
2. Update item (state2)
3. Delete item
4. Sync with the server

Actual Result:
In the store we have the item (state1).
Expected result:
Item should be removed (from both the cache and the backend).

#### Changes
`DELETE` sync request will be added to a stack of sync requests even if `SAVE` request already exists in the stack.

#### Tests
Instrumented

### CODE REVIEW

File: android-lib/src/androidTest/java/com/kinvey/androidTest/store/DataStoreTest.java [hunk 1]
```diff
@@ -3246,4 +3246,30 @@ public void testUpdateInternalList() throws InterruptedException, IOException {
...
+        assertNotNull(callback.result.getUsername());
+        assertEquals(TEST_USERNAME_2, callback.result.getUsername());
+
+        DefaultKinveyDeleteCallback deleteCallback = delete(store, callback.result.getId(), DEFAULT_TIMEOUT);
+        assertNotNull(deleteCallback.result);
+
+        sync(store, DEFAULT_TIMEOUT);
+
+        assertEquals(0, find(store, client.query().equals(Constants._ID, callback.result.getId()), DEFAULT_TIMEOUT).result.size());
+    }
```
Comment (vinaygahlawat) [inline android-lib/src/androidTest/java/com/kinvey/androidTest/store/DataStoreTest.java]: Please also add an assertion here to make sure that there sync queue is cleared out.

File: java-api-core/src/com/kinvey/java/store/requests/data/PushRequest.java [hunk 2]
```diff
@@ -76,8 +79,14 @@ public Void execute() throws IOException {
                             break;
                     }
                 }
-                syncManager.executeRequest(client, syncRequest);
-                syncManager.deleteCachedItem((String) syncItem.get("_id"));
+                try {
+                    syncManager.executeRequest(client, syncRequest);
+                } catch (Exception e) {
+                    if (!e.getMessage().contains(IGNORED_EXCEPTION)) {
```
[the remark this answers is not in the document]
Comment (yuliya-guseva) [reply java-api-core/src/com/kinvey/java/store/requests/data/PushRequest.java]: Clarified exception and response code for push ignoring.

File: android-lib/src/main/java/com/kinvey/android/async/CallableAsyncPushRequestHelper.java [hunk 3]
```diff
@@ -26,8 +29,14 @@
 
     @Override
     public Long call() throws Exception {
-        manager.executeRequest(client, syncRequest);
-        manager.deleteCachedItem((String) syncItem.get("_id"));
+        try {
+            manager.executeRequest(client, syncRequest);
+        } catch (Exception e) {
+            if (!e.getMessage().contains(IGNORED_EXCEPTION)) {
```
[the remark this answers is not in the document]
Comment (yuliya-guseva) [reply android-lib/src/main/java/com/kinvey/android/async/CallableAsyncPushRequestHelper.java]: Clarified exception and response code for push ignoring.

File: android-lib/src/androidTest/java/com/kinvey/androidTest/store/DataStoreTest.java (hunk 1 above)
Comment (yuliya-guseva) [reply android-lib/src/androidTest/java/com/kinvey/androidTest/store/DataStoreTest.java]: Added

File: java-api-core/src/com/kinvey/java/sync/SyncManager.java [hunk 4]
```diff
@@ -144,7 +144,10 @@ public void enqueueRequest(String collectionName, AbstractKinveyJsonClientReques
         entityQuery.equals(META_DOT_ID, id);
         List<SyncItem> itemsList = requestCache.get(entityQuery);
 
-        if (itemsList.size() == 0) {
+        if (itemsList.isEmpty()) {
+            requestCache.save(createSyncItem(collectionName, method, networkManager, id));
+        } else if (method == RequestMethod.DELETE) {
+            requestCache.delete(entityQuery);
             requestCache.save(createSyncItem(collectionName, method, networkManager, id));
```
Comment (vinaygahlawat) [inline java-api-core/src/com/kinvey/java/sync/SyncManager.java]: Does similar logic have to be implemented for `public <T extends GenericJson> void enqueueRequests(...)` methods?

Comment (yuliya-guseva) [reply java-api-core/src/com/kinvey/java/sync/SyncManager.java]: Yes, it has to be implemented for `enqueueRequests` as well. Updated.

**Pull request merged**